src/EditorFeatures/CSharpTest/Completion/CompletionProviders/AwaitCompletionProviderTests.cs

```diff
@@ -0,0 +1,182 @@
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TestExpressionInAsyncMethod(bool topLevelStatement)
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"var z = $$", isAsync: true, topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TestUsingStatement(bool topLevelStatement)
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"using $$", topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestUsingDirective()
+            => await VerifyAbsenceAsync("using $$");
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TestForeachStatement(bool topLevelStatement)
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"foreach $$", topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TestNotInQuery(bool topLevelStatement)
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"var z = from a in ""char""
+          select $$", isAsync: true, topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [WorkItem(907052, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/907052")]
+        [Theory]
+        [CombinatorialData]
+        public async Task TestInFinally(bool topLevelStatement)
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"try { }
+finally { $$ }", isAsync: true, topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [WorkItem(907052, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/907052")]
+        [Theory]
+        [CombinatorialData]
+        public async Task TestInCatch(bool topLevelStatement)
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"try { }
+catch { $$ }", isAsync: true, topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TestNotInLock(bool topLevelStatement)
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(
+@"lock(this) { $$ }", isAsync: true, topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TestInAsyncLambdaInCatch(bool topLevelStatement)
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"try { }
+catch { var z = async () => $$ }", isAsync: true, topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TestAwaitInLock(bool topLevelStatement)
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"lock($$", isAsync: true, topLevelStatement: topLevelStatement), LanguageVersion.CSharp9);
```

Comment:
i really disliek this stuff as i honestly cannot make out waht the test is testing anymore.   i like declarative tests wher ei look at the input and that's that.

Reply:
@CyrusNajmabadi I strongly agree. This is one of the places I think duplicate code makes sense more than extracting a helper. This was just a copy from the existing tests. I'll fixup this.

Reply:
@CyrusNajmabadi Done